lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -296,16 +296,25 @@ public Thread StartTask(IConfiguration taskConfiguration)
 
                 var taskInjector = _contextInjector.ForkInjector(taskConfiguration);
 
-                var taskRuntime = taskInjector.GetInstance<TaskRuntime>();
-
                 try
                 {
+                    var taskRuntime = taskInjector.GetInstance<TaskRuntime>();
                     _task = Optional<TaskRuntime>.Of(taskRuntime);
                     return taskRuntime.RunTask();
                 }
                 catch (Exception e)
                 {
-                    var ex = new TaskClientCodeException(string.Empty, Id, "Unable to run the new task", e);
+                    var taskId = string.Empty;
+                    try
+                    {
+                        taskId = taskInjector.GetNamedInstance<TaskConfigurationOptions.Identifier, string>();
+                    }
+                    catch (Exception)
+                    {
+                        LOGGER.Log(Level.Error, "Unable to get Task ID from TaskConfiguration.");
+                    }
+
+                    var ex = TaskClientCodeException.Create(taskId, Id, "Unable to run the new task", e);
```

Comment:
I believe this design was copied from Java, which is used to indicate when we are not able to start Tasks. After a Task starts, the Exception will no longer be wrapped in `TaskClientCodeException`. You are also able to distinguish the actual Exception by inspecting the inner Exception.

In other words, `TaskClientCodeException` will only be returned when certain control actions are not implemented correctly, e.g. Start, Stop, and Suspend.